### PR TITLE
Add additional agent configuration and the blueocean plugin

### PR DESCRIPTION
**Pull request by berndtj**
The additional configuration values allow more customizable deployments and support running docker in docker on the agents (via mounted docker.sock).  Additionally this adds the blueocean plugin which adds a modern UI and github pipeline support.

### CODE REVIEW

**Comment by berndtj** [comment]
CLA has been signed now.

File: stable/jenkins/templates/config.yaml [hunk 1]
```diff
@@ -37,7 +37,14 @@ data:
               <idleMinutes>0</idleMinutes>
               <label></label>
               <nodeSelector></nodeSelector>
-              <volumes/>
+              <volumes>
+{{- range $key, $val := .Values.Agent.Volumes }}
```
Comment (viglesiasce) [inline stable/jenkins/templates/config.yaml]: Can you implement this as was done in #981?

**Review by viglesiasce — COMMENTED** [review]
Please make the volume type configurable and bump the Chart version

File: stable/jenkins/templates/config.yaml (hunk 1 above)
Comment (berndtj) [reply stable/jenkins/templates/config.yaml]: Ha, didn't see that PR.  Sure, I can do that.

**Comment by berndtj** [comment]
Also bumped chart version, per request

**Comment by berndtj** [comment]
I changed my remote repo.  I will close this and create a new pr.

**Pull request closed without merging**